### ISSUE TITLE
What versions of OpenSSL3 are FIPS certified? 

### ISSUE THREAD

**Issue by Bonn93** [openssl/openssl#20239]
I am looking to use OpenSSL3 with FIPS and have been compiling some versions and following the FIPS install and config options. But I am unable to find exactly which versions are "certified for FIPS". 

The FIPS certificate states 3.0.0 is certified here https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4282

Does this mean that 3.0.1/2/3 etc are not certified? 

Thanks :)

**Comment by paulidale** [comment]
The FIPS provider from OpenSSL 3.0.0 is certified.  That's all currently.

This provider can be used with any OpenSSL 3.0 release.

So, build OpenSSL 3.0.0 to get the FIPS provider.  Build 3.0.8 without FIPS and then use the 3.0.0 FIPS provider with it.

**Comment by paulidale** [comment]
Configure, build and install OpenSSL 3.0.0 exactly as per the [security policy](https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp4282.pdf).

Then configure, build and install OpenSSL 3.0.8 without the `enable-fips` option.

**Comment by slontis** [comment]
Examples of provider related files are fips.so & legacy.so
libcrypto.so includes inbuilt providers for default and base.

So Use 3.0.0 code to produce the fips.so, and then use 3.0.X to build libcrypto and legacy.

**Comment by Jayukie** [comment]
I found that some patches of FIPS 140-3 have been included in openssl 3.1. 
Can I build OpenSSL 3.1 to get the FIPS provider, and build 3.0.4 without FIPS, and then use the 3.1 FIPS provider with it ?

**Comment by paulidale** [comment]
No, 3.1 is not validated.  It's not even been submitted for validation at this stage.

**Comment by Bonn93** [comment]
Hey, I'd like to re-open this, the docs now suggest otherwise, and the CMVP/Cert issued links 3.0.8, is 3.0.0 and 3.0.8 valid?
